src/executor/attach_op_execs_pass.cc

```diff
@@ -40,6 +40,11 @@ const OperatorProperty* OpPropGetOpProperty(const NodeAttrs& attrs);
 
 namespace exec {
 
+class MKLDNNOpExecutor : public OpExecutor {
+ protected:
+  std::vector<NDArray> in_array_fallback;
```

Comment:
Why do we need an extra copy rather than just editing the in_array directly?

Reply:
there is a race condition if we attempt to reorder the `read_var` in place. other operators may be trying to read from it the same time (since they are expected to be read only consts)

Reply:
If that's the case, why do we even bother to create a subclass with this extra data member?